packages/eas-cli/src/commands/branch/delete.ts

```diff
@@ -87,17 +88,18 @@ export default class BranchDelete extends EasCommand {
     },
   ];
   static flags = {
-    json: Flags.boolean({
-      description: `return JSON with the edited branch's ID and name.`,
-      default: false,
-    }),
+    // do not expose limit/offset since they don't make sense on this command
+    json: EasPaginatedQueryFlags.json,
+    'non-interactive': EasPaginatedQueryFlags['non-interactive'],
   };
 
   async runAsync(): Promise<void> {
     let {
-      args: { name },
-      flags: { json: jsonFlag },
+      args: { name: branchName },
+      flags,
     } = await this.parse(BranchDelete);
+    const paginatedQueryOptions = getPaginatedQueryOptions(flags);
+    const { json: jsonFlag, 'non-interactive': nonInteractiveFlag } = flags;
```

Comment:
if both flags exist in other places `--json` can only be passed if `--non-interactive` is already set.

It would be good to extract it to some common util, e.g. you could check that in getPaginatedQueryOptions (you should rename it to sth more appropriate in that case)

Reply:
The same applies to other commands from this PR and some form the previous one.

Reply:
Also you are using inconsistent naming, sometimes it's json/nonInteractive and sometimes jsonFlag/nonInteractiveFlag

Reply:
hey! so, lmk if I'm misunderstanding your comment:

> if both flags exist in other places --json can only be passed if --non-interactive is already set.

Are you saying that we want to explicitly not support json output for interactive commands? In that case, I think we should be making use of the `dependsOn` property that ocliff flags provides.

Reply:
> Are you saying that we want to disallow json output for commands run in interactive mode?

Yes, that is how all commands with both nonInteractive and json flags worked before.

If there is any prompt in json only output it would not work correctly. Another alternative would be to implicitly set nonInteractive if json flag is present, I'm fine with this approach too, but you would need to also change all the existing commands that have both of those flags.

> Sure. So my intention was that every line of code I've touched now uses the xFlag convention

In more complicated commands we usually build some context/options object based on flag and other input. This convention might be problematic because you can name sth nonInteractiveFlag when operating inside top-level command code, but for example inside some util function or as part of the context object this name would not be correct and you would have to use `nonInteractive`. As a result, you will have a bunch of `nonInteractive: nonInteractiveFlag` in your code. If this separation is intentional, I'm fine with it.

Reply:
re `xFlag`: my motivation was aid readability by making the variable name more explicit. In the scenario you're talking about where you pass some collection of flags to a utility method (`commands/build/index.ensurePlatformSelected`) i think it's already pretty explicit that your dealing with flags because there is a `flag.X` syntax going on.

I don't have a strong preference for one way or the other, so if you prefer `x` over `xFlag` for top-level command code I'm happy to stick with that. When I started working on the code-base there was a mix of both, so I just chose one approach and stuck with it.